docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+Hypothesis CSS Style Guide
+==========================
+
+Hypothesis' CSS generally follows the principles of [Trello's CSS Style Guide](https://github.com/trello/trellisheets)
+for writing maintainable styling. In particular:
+
+ * Use simple, non-nested class selectors where possible
+ * Organize CSS into components. Unlike Trello, we use BEM-style naming (see below) and do not indent selectors
+ * Be careful with CSS preprocessor features. Unlike Trello, we use SASS but the principle
+   of using 'vanilla' CSS and being careful with preprocessor features still applies.
+   See the note about `@extends` below
+ * Use naming conventions to indicate which classes are purely presentational and which are referenced in code
```

Comment:
This is referring to the use of `js-*` classes for JavaScript to find particular elements, and `is-*` classes for state that's added and removed by JavaScript? If so I think this sentence would be clearer if concrete, something like: use `js-*` classes to enable JavaScript code to find particular DOM elements, and `is-*` classes for states that are added and removed by JavaScript." And then maybe link to the section(s) that explain this in more detail.